### PR TITLE
[PHI decoupling]migrate enforce_custom.h from fluid to phi

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### PR types
Others

### PR changes
Others

### Describe
migrate enforce_custom.h from fluid to phi
[used AI Studio]

- #47615

### CODE REVIEW

**Comment by luotao1** [comment]
@YuanRisheng 可以进行 review

File: paddle/phi/core/enforce_custom.h [hunk 1]
```diff
@@ -16,12 +16,11 @@ limitations under the License. */
 #ifdef PADDLE_WITH_CUSTOM_DEVICE
 #include <string>
 
-#include "paddle/fluid/platform/enforce.h"
 #include "paddle/phi/backends/device_ext.h"
+#include "paddle/phi/core/enforce.h"
```
Comment (YuanRisheng) [inline paddle/phi/core/enforce_custom.h]: 把这个文件放置到backends/custom下吧

File: paddle/phi/core/enforce_custom.h [hunk 2]
```diff
@@ -16,12 +16,11 @@ limitations under the License. */
...
 
-namespace paddle {
-namespace platform {
-namespace details {
+namespace phi {
+// namespace details {
```
Comment (YuanRisheng) [inline paddle/phi/core/enforce_custom.h]: 删除多余注释

File: paddle/phi/core/enforce_custom.h [hunk 3]
```diff
@@ -33,27 +32,25 @@ struct CustomDeviceStatusType {};
   }
 
 DEFINE_CUSTOM_DEVICE_STATUS_TYPE(C_Status, C_SUCCESS);
-}  // namespace details
+// }  // namespace details
```
Comment (YuanRisheng) [inline paddle/phi/core/enforce_custom.h]: 同上

File: paddle/phi/core/enforce_custom.h (hunk 2 above)
Comment (Asthestarsfalll) [reply paddle/phi/core/enforce_custom.h]: done

File: paddle/phi/core/enforce_custom.h (hunk 3 above)
Comment (Asthestarsfalll) [reply paddle/phi/core/enforce_custom.h]: done

**Comment by Asthestarsfalll** [comment]
@YuanRisheng CI已通过

**Review by YuanRisheng — APPROVED** [review]

**Pull request merged**